app/models/manageiq/providers/ibm_cloud/power_virtual_servers/cloud_manager/provision_workflow.rb

```diff
@@ -176,6 +182,21 @@ def validate_ip_address(_field, _values, _dlg, _fld, value)
     return _('IP-address field has to be either blank or a valid IPv4 address') unless valid
   end
 
+  def validate_placement_group(_filed, _values, _dlg, _fld, value)
+    return if value.blank?
+
+    placement_group = ar_ems.placement_groups.find_by!(:ems_ref => value)
+    vms_in_placement_group = placement_group.vms
+
+    # If policy is affinity, check to make sure the new vm has the same flavor as all the members of the group
+    valid = if placement_group[:policy] == 'affinity' && vms_in_placement_group.present?
+              vms_in_placement_group.first.flavor.name == values&.dig(:sys_type, 1)
+            else
+              true
+            end
+    return _('Invalid placement group - incompatible colocation policy') unless valid
```

Comment:
You actually don't need this redundant `return` here as ruby automatically returns the last statement

```suggestion
    _('Invalid placement group - incompatible colocation policy') unless valid
```